src/prism/params/errors.cr

```diff
@@ -1,68 +1,86 @@
 module Prism::Params
+  abstract class ParamError < Exception
+    getter param
+
+    def initialize(@param : AbstractParam)
+      super(build_message)
+    end
+
+    def path
+      param.path + [param.name]
```

Comment:
Change to `param.path.push(param.name)` to avoid array initialization

Reply:
doesn't that mutate `param.path`?